bundles/org.openhab.binding.qbus/src/main/java/org/openhab/binding/qbus/internal/handler/QbusThermostatHandler.java

```diff
@@ -0,0 +1,244 @@
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        QbusCommunication QComm = getCommunication("Thermostat", thermostatId);
+
+        if (QComm == null) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "Bridge communication not initialized when trying to execute command for thermostat "
+                            + thermostatId);
+            return;
+        }
+
+        Map<Integer, QbusThermostat> thermostatComm = QComm.getThermostat();
+
+        if (thermostatComm != null) {
+
+            QbusThermostat QThermostat = thermostatComm.get(thermostatId);
+
+            if (QThermostat == null) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                        "Bridge communication not initialized when trying to execute command for Scene "
+                                + thermostatId);
+                return;
+            } else {
+                scheduler.submit(() -> {
+                    if (!QComm.communicationActive()) {
+                        restartCommunication(QComm, "Thermostat", thermostatId);
+                    }
+
+                    if (QComm.communicationActive()) {
+                        if (command == REFRESH) {
+                            handleStateUpdate(QThermostat);
+                            return;
+                        }
+
+                        switch (channelUID.getId()) {
+                            case CHANNEL_MODE:
+                                handleModeCommand(QThermostat, command);
+                                break;
+
+                            case CHANNEL_SETPOINT:
+                                handleSetpointCommand(QThermostat, command);
+                                break;
+
+                        }
+                    }
+                });
+            }
+        }
+    }
+
+    /**
+     * Puts thing offline
+     *
+     * @param message
+     */
+    public void thingOffline(String message) {
+        updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.COMMUNICATION_ERROR, message);
+    }
+
+    /**
+     * Executes the Mode command
+     */
+    private void handleModeCommand(QbusThermostat QThermostat, Command command) {
+        String snr = getSN();
+        if (command instanceof DecimalType) {
+            int mode = ((DecimalType) command).intValue();
+            if (snr != null) {
+                QThermostat.executeMode(mode, snr);
+            } else {
+                thingOffline("No serial number configured for  " + thermostatId);
+            }
+        }
+    }
+
+    /**
+     * Executes the Setpoint command
+     */
+    private void handleSetpointCommand(QbusThermostat QThermostat, Command command) {
+        String snr = getSN();
+        if (command instanceof QuantityType<?>) {
+            QuantityType<?> s = (QuantityType<?>) command;
+            double sp = s.doubleValue();
```

Comment:
You need to convert the temperature value you expect. If the user has configured Fahrenheit, `s.doubleValue()` would return the Fahrenheit value. You can call `c.toUnit()` for the conversion.